sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuBringBackToHost.scala

```diff
@@ -46,7 +46,10 @@ case class GpuBringBackToHost(child: SparkPlan) extends ShimUnaryExecNode with G
       try {
         val hostColumns = (0 until b.numCols()).map(
           i => b.column(i).asInstanceOf[GpuColumnVector].copyToHost())
-        new ColumnarBatch(hostColumns.toArray)
+        if (hostColumns.isEmpty) {
+          throw new RuntimeException("hostColumns should not be empty")
+        }
```

Comment:
I presume that the contract for `GpuExec.internalDoExecuteColumnar()` is that its input should have at least one column for non-leaf nodes. Let me know if this is not true.

Reply:
Just simplified this code to directly get the row count from the input batch instead of `hostColumns`.